client/my-sites/domains/email-providers-upsell/index.jsx

```diff
@@ -4,17 +4,21 @@ import { useTranslate } from 'i18n-calypso';
 import page from 'page';
 import { useSelector } from 'react-redux';
 import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
-import { domainAddNew } from 'calypso/my-sites/domains/paths';
+import { domainAddEmailUpsell, domainAddNew } from 'calypso/my-sites/domains/paths';
 import EmailProvidersComparison from 'calypso/my-sites/email/email-providers-comparison';
 import EmailProvidersStackedComparison from 'calypso/my-sites/email/email-providers-stacked-comparison';
 import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
 
-export default function EmailProvidersUpsell( { domain } ) {
+export default function EmailProvidersUpsell( { domain, selectedIntervalLength } ) {
 	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
 	const translate = useTranslate();
 
 	const comment = '%(domainName)s is the domain name, e.g example.com';
 
+	const changeIntervalLength = ( props, intervalLength ) => {
+		page( domainAddEmailUpsell( selectedSiteSlug, props.selectedDomainName, intervalLength ) );
+	};
```

Comment:
I suspect we should use the same variable name than in the other handler to stay consistent:

```suggestion
	const changeIntervalLength = ( props, newIntervalLength ) => {
		page( domainAddEmailUpsell( selectedSiteSlug, props.selectedDomainName, newIntervalLength ) );
	};
```